<?xml version="1.0"?>
<launch>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="true"/>
  <arg name="ticks_meter" default="3185" />
  <arg name="motor_direction" default="1" />
  <arg name="twist_factor" default="542" />
  <arg name="xacro_path" />
  <arg name="params_dir_path" />
  <!--include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" /-->


  <param name="robot_description" command="$(find xacro)/xacro.py '$(arg xacro_path)'" />
  <!-- send fake joint values -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="use_gui" value="False"/>
  </node>


  <!--Robot State Publisher-->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/> 
  <!--include file="$(find model_bringup)/params/amcl.launch.xml"/-->
  <include file="$(find rplidar_ros)/launch/rplidar.launch"/>
  <!--node name="self_localize_1" pkg="model_control" type="self_localize_1.py" output="screen"/-->
  <!--node name="micvision_localization_node" pkg="micvision" type="micvision_localization_node" output="screen"/-->
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
    <param name="port" value="/dev/arduino"/>
    <param name="baud" value="115200"/>
  </node>
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
            <!--rosparam file="$(arg params_dir_path)/costmap_common_params.yaml" command="load" ns="global_costmap" /-->
            <!--rosparam file="$(arg params_dir_path)/costmap_common_params.yaml" command="load" ns="local_costmap" /-->
            <!--rosparam file="$(arg params_dir_path)/local_costmap_params.yaml" command="load" /-->
            <rosparam file="$(arg params_dir_path)/global_costmap_params.yaml" command="load" />
            <rosparam file="$(arg params_dir_path)/base_local_planner_params.yaml" command="load" />
            <rosparam file="$(arg params_dir_path)/move_base_params.yaml" command="load" />
  </node>
    <!--<param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />-->

        <!-- convert provide equal cmd_vel to both motors-->
        <node pkg="model_control" type="test.py" name="twist_motor" output="screen">
            <rosparam param="K" subst_value="True">$(arg twist_factor)</rosparam>
        </node>

        <!-- Publish Odom and Broadcast TF-->
        <node pkg="model_control" type="diff_tf.py" name="diff_tf" output="screen">
            <rosparam param="ticks_meter" subst_value="True">$(arg ticks_meter)</rosparam>
            <rosparam param="motor_direction" subst_value="True">$(arg motor_direction)</rosparam>
        </node>


  <node pkg="gmapping" type="slam_gmapping" name="mapping"/>
      <param name="xmin" value="-25.0"/>
      <param name="ymin" value="-25.0"/>
      <param name="xmax" value="25.0"/>
      <param name="ymax" value="25.0"/>

</launch>

